docs/aws/audit/kubernetesmonitoring/rules/is_version_recent_remediation.mdx

```diff
@@ -1,6 +1,224 @@
+```
+
+### Summary
+1. **Install Required Libraries**: Ensure `boto3` and `requests` are installed.
+2. **Fetch the Latest Stable Kubernetes Version**: Use `requests` to get the latest version.
+3. **List Existing EKS Clusters and Their Versions**: Use `boto3` to list clusters and their versions.
+4. **Compare and Update Clusters**: Compare current versions with the latest and update if necessary.
+
+By following these steps, you can automate the process of ensuring your EKS clusters are always using the latest stable version of Kubernetes.
+</Accordion>
+
+</AccordionGroup>
+</Tab>
+<Tab title='Cause'>
+### Check Cause
+<AccordionGroup>
+<Accordion title='Using Console' defaultOpen='true'>
+1. Log in to the AWS Management Console and navigate to the Amazon EKS service.
+2. In the left navigation pane, select "Clusters" to open the list of your existing EKS clusters.
+3. Click on the name of the cluster that you want to check. This will open the cluster's details page.
+4. In the "Details" section, look for the "Kubernetes version" field. Compare this version with the latest stable version of Kubernetes available. If the versions do not match, it indicates a misconfiguration.
+</Accordion>
+
+<Accordion title='Using CLI'>
+1. First, you need to install and configure AWS CLI on your local machine. Make sure you have the necessary permissions to access the EKS clusters.
+
+2. Once the AWS CLI is set up, you can list all the EKS clusters in your account using the following command:
+   ```
+   aws eks list-clusters
+   ```
+   This command will return a list of all the EKS clusters in your account.
+
+3. To get the version of Kubernetes running on a specific EKS cluster, use the following command:
+   ```
+   aws eks describe-cluster --name <cluster-name> --query "cluster.version"
+   ```
+   Replace `<cluster-name>` with the name of your EKS cluster. This command will return the version of Kubernetes running on the specified EKS cluster.
+
+4. To check the latest stable version of Kubernetes, you can visit the Kubernetes GitHub page or use the following command:
+   ```
+   curl -s https://storage.googleapis.com/kubernetes-release/release/stable.txt
+   ```
+   Compare the version obtained from the EKS cluster with the latest stable version. If they are not the same, it means the EKS cluster is not using the latest stable version of Kubernetes.
+</Accordion>
+
+<Accordion title='Using Python'>
+1. Install the necessary Python libraries: You will need the boto3 library, which is the Amazon Web Services (AWS) SDK for Python. This allows Python developers to write software that makes use of services like Amazon S3, Amazon EC2, and others. You can install it using pip:
+
+   ```python
+   pip install boto3
+   ```
+
+2. Import the necessary libraries and establish a session with AWS:
+
+   ```python
+   import boto3
+   session = boto3.Session(
+       aws_access_key_id='YOUR_ACCESS_KEY',
+       aws_secret_access_key='YOUR_SECRET_KEY',
```

Comment:
<b>Secret:</b> Secret Keyword <br/>
                    <b><a href="https://console.cloudanix.com/code?t=secrets" target='_blank'>Click here for more details</a></b>